nucleus/grizzly/config/src/main/java/org/glassfish/grizzly/config/GenericGrizzlyListener.java

```diff
@@ -739,95 +738,36 @@ protected void configureHttpProtocol(final ServiceLocator habitat,
-                Object mode = ((isNpnMode) ? enumConstants[1] : enumConstants[0]);
-                spdyAddon = loadAddOn("org.glassfish.grizzly.spdy.SpdyAddOn", new Class[]{spdyMode}, mode);
-            }
-
-            if (spdyAddon != null) {
-                // Configure SpdyAddOn
-                configureElement(locator, listener, spdyElement, spdyAddon);
-                
-                // Spdy requires access to more information compared to the other addons
-                // that are currently leveraged.  As such, we'll need to mock out a
-                // Grizzly NetworkListener to pass to the addon.  This mock object will
-                // only provide the information necessary for the addon to operate.
-                // It will be important to keep this mock in sync with the details the
-                // addon requires.
-                spdyAddon.setup(createMockListener(secure), builder);
-                isSpdyEnabled = true;
-            }
-        }
-    }
-
     protected void configureHttp2Support(final ServiceLocator locator,
                                         final NetworkListener listener,
-                                        final Http2 http2Element,
+                                        final Http httpElement,
                                         final FilterChainBuilder builder,
                                         final boolean secure) {
-        if (!skipHttp2 && http2Element != null && GrizzlyConfig.toBoolean(http2Element.getEnabled())) {
-
-            // first try to lookup a service appropriate for the mode
-            // that has been configured.
-            AddOn http2Addon = locator.getService(AddOn.class, "http2");
-
-            // if no service was found, attempt to load via reflection.
-            if (http2Addon == null) {
-                http2Addon = loadAddOn("org.glassfish.grizzly.http2.Http2AddOn", new Class[]{});
-            }
-
-            if (http2Addon != null) {
-                // Configure SpdyAddOn
-                configureElement(locator, listener, http2Element, http2Addon);
-                
-                // Spdy requires access to more information compared to the other addons
-                // that are currently leveraged.  As such, we'll need to mock out a
-                // Grizzly NetworkListener to pass to the addon.  This mock object will
-                // only provide the information necessary for the addon to operate.
-                // It will be important to keep this mock in sync with the details the
-                // addon requires.
-                http2Addon.setup(createMockListener(secure), builder);
-                isHttp2Enabled = true;
-            }
+        if (!skipHttp2 && httpElement != null && httpElement.isHttp2Enabled()) {
+
+            Http2AddOn http2Addon = new Http2AddOn(Http2Configuration.builder()
+                    .disableCipherCheck(httpElement.isHttp2DisableCipherCheck())
```

Comment:
Can these configuration properties be added to the administration console and be made configurable via an asadmin command.